stdlib/public/core/Array.swift

```diff
@@ -1952,26 +1952,6 @@ extension Array where Element : Differentiable {
 
     public typealias TangentVector =
       Array<Element.TangentVector>.DifferentiableView
-    public typealias AllDifferentiableVariables =
```

Comment:
In a separate patch, `Array.DifferentiableView` should be replaced with a `struct TangentVector` [as described in the design overview](https://docs.google.com/document/d/1bPepWLfRQa6CtXqKA8CDQ87uZHixNav-TFjLSisuKag/edit#bookmark=id.vnijnoui0irg).